### ISSUE TITLE
Netlify Adapter does not return properly formed response for binary data

### ISSUE THREAD

**Issue by bummzack** [sveltejs/kit#4174]
### Describe the bug

In order to return binary data (eg. a generated Image) from a Netlify function, the data must be converted to a `base64` encoded string with the proper content-type.

This is done in [`handler.js`](https://github.com/sveltejs/kit/blob/master/packages/adapter-netlify/src/handler.js#L21), but the condition that checks for `Uint8Array` will always fail, as the response type changed with #3384 

From what I can tell, there is currently no way to determine the "type" from the response, other than looking at the `content-type` header and using something like [`is_text`](https://github.com/sveltejs/kit/blob/827d1623d64a9886032c8bfc8e2315420d1599d0/packages/kit/src/runtime/server/endpoint.js#L30)

### Reproduction

Here's a minimal setup: https://github.com/bummzack/sveltekit-netlify-binary-response

There's twice the same code running in a SvelteKit Endpoint ([`image.png`](https://github.com/bummzack/sveltekit-netlify-binary-response/blob/main/src/routes/image.png.js)) and a separate [Netlify function](https://github.com/bummzack/sveltekit-netlify-binary-response/blob/main/netlify/functions/test.js).

The SvelteKit endpoint does not encode the image properly and it won't be displayed: 
https://elated-knuth-7439db.netlify.app/image.png

The direct implementation as a separate Netlify function works as expected:
https://elated-knuth-7439db.netlify.app/.netlify/functions/test

### System Info

```shell
Local system is irrelevant, since the issue doesn't occur locally.

  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.30 
    @sveltejs/kit: next => 1.0.0-next.288 
    svelte: ^3.44.0 => 3.46.4
```

### Severity

annoyance

**Comment by bummzack** [comment]
Sorry, I wanted to create a PR for this, but I banged my head against the keyboard while trying to understand how I could import `is_text` from `kit/src/runtime/server/endpoint.js` into the handler function… 

I guess it somehow needs to be exported from `kit`, but the whole build/setup/pipeline eludes me.

**Comment by goynang** [comment]
I'm having the same problem on Netlify when trying to serve out a dynamically generated PDF. Works fine locally.

I tried to create a PR for the `is_text` fix that @bummzack mentions but hit the same issues. I tried good old fashion copy/paste but then got lost within the darker arts of npm trying to get it to run my forked version (from the monorepo package) and deploy.

This does look like it should be a relatively simple thing to fix for anyone that knows what they are doing (unlike myself!).

I will run a custom netlify function in the meantime as a workaround.
